src/lib/components/ui/Modal/Modal.jsx

```diff
@@ -111,6 +111,7 @@ class Modal extends React.Component {
                 label={action.label}
                 loadingIcon={action.loadingIcon}
                 variant={action.variant}
+                type={action.type || 'button'}
```

Comment:
I think the default value should be handled in `defaultProps`

Reply:
It is not supported for nested data.

Reply:
True, I did not realize that it is nested.
How about:

`{...action.type && { type: action.type } }`

Although functionally same, I like it a little bit more because it does not provide default value at an unexpected place. Do as you see fit though…